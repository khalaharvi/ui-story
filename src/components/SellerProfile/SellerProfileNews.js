import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import RelatedModuleTemplate from 'templates/RelatedModule';

import SellerProfileNewsItem from './SellerProfileNewsItem';
import { cx } from './SellerProfile';

const SellerProfileNews = ({ news, newsComponent: NewsComponent }) => (
  <RelatedModuleTemplate className={cx('news')} title="News">
    <ul className="columns columns--4x">
      {
        news.map(n => (
          <li key={n.id} className={classnames('col', cx('news-item'))}>
            <NewsComponent news={n} />
          </li>
        ))
      }
    </ul>
  </RelatedModuleTemplate>
);

SellerProfileNews.propTypes = {
  newsComponent: PropTypes.func,
  news: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
    image: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired
  })).isRequired
};

SellerProfileNews.defaultProps = {
  newsComponent: props => <SellerProfileNewsItem {...props} />
};

export default SellerProfileNews;
