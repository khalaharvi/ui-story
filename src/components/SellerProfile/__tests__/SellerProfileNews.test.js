import React from 'react';
import { shallow } from 'enzyme';

import SellerProfileNews from '../SellerProfileNews';

const news = [
  {
    id: 1,
    image: 'https://cdn.deringhall.com/announcement/image/aa327cae-518b-489a-bf6b-ff1c040c5fa0_elle.decor.jpg?auto=format&fit=max&q=60&w=670',
    title: 'Elle Decor',
    type: 'Announcement'
  },
  {
    id: 2,
    image: 'https://cdn.deringhall.com/images/2302/original/image.jpg?auto=format&fit=max&q=60&w=670',
    title: 'Architectural Digest',
    type: 'Announcement'
  },
  {
    id: 3,
    image: 'https://cdn.deringhall.com/images/718/original/cover.jpg?auto=format&fit=max&q=60&w=670',
    title: 'Shining Through',
    type: 'Press'
  },
  {
    id: 4,
    image: 'https://cdn.deringhall.com/images/681/original/cover.jpg?auto=format&fit=max&q=60&w=670',
    title: 'Remote Possibilities',
    type: 'Press'
  }
];

describe('<SellerProfileNews />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <SellerProfileNews news={news} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with newsComponent', () => {
    const wrapper = shallow(
      <SellerProfileNews
        news={news}
        newsComponent={({ news: n }) => <p>{n.title}</p>}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
