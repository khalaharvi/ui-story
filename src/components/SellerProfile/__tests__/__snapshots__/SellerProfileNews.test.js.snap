// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SellerProfileNews /> renders correctly 1`] = `
<RelatedModuleTemplate
  className="dh-seller-profile-news"
  title="News"
>
  <ul
    className="columns columns--4x"
  >
    <li
      className="col dh-seller-profile-news-item"
      key="1"
    >
      <newsComponent
        news={
          Object {
            "id": 1,
            "image": "https://cdn.deringhall.com/announcement/image/aa327cae-518b-489a-bf6b-ff1c040c5fa0_elle.decor.jpg?auto=format&fit=max&q=60&w=670",
            "title": "Elle Decor",
            "type": "Announcement",
          }
        }
      />
    </li>
    <li
      className="col dh-seller-profile-news-item"
      key="2"
    >
      <newsComponent
        news={
          Object {
            "id": 2,
            "image": "https://cdn.deringhall.com/images/2302/original/image.jpg?auto=format&fit=max&q=60&w=670",
            "title": "Architectural Digest",
            "type": "Announcement",
          }
        }
      />
    </li>
    <li
      className="col dh-seller-profile-news-item"
      key="3"
    >
      <newsComponent
        news={
          Object {
            "id": 3,
            "image": "https://cdn.deringhall.com/images/718/original/cover.jpg?auto=format&fit=max&q=60&w=670",
            "title": "Shining Through",
            "type": "Press",
          }
        }
      />
    </li>
    <li
      className="col dh-seller-profile-news-item"
      key="4"
    >
      <newsComponent
        news={
          Object {
            "id": 4,
            "image": "https://cdn.deringhall.com/images/681/original/cover.jpg?auto=format&fit=max&q=60&w=670",
            "title": "Remote Possibilities",
            "type": "Press",
          }
        }
      />
    </li>
  </ul>
</RelatedModuleTemplate>
`;

exports[`<SellerProfileNews /> renders correctly with newsComponent 1`] = `
<RelatedModuleTemplate
  className="dh-seller-profile-news"
  title="News"
>
  <ul
    className="columns columns--4x"
  >
    <li
      className="col dh-seller-profile-news-item"
      key="1"
    >
      <newsComponent
        news={
          Object {
            "id": 1,
            "image": "https://cdn.deringhall.com/announcement/image/aa327cae-518b-489a-bf6b-ff1c040c5fa0_elle.decor.jpg?auto=format&fit=max&q=60&w=670",
            "title": "Elle Decor",
            "type": "Announcement",
          }
        }
      />
    </li>
    <li
      className="col dh-seller-profile-news-item"
      key="2"
    >
      <newsComponent
        news={
          Object {
            "id": 2,
            "image": "https://cdn.deringhall.com/images/2302/original/image.jpg?auto=format&fit=max&q=60&w=670",
            "title": "Architectural Digest",
            "type": "Announcement",
          }
        }
      />
    </li>
    <li
      className="col dh-seller-profile-news-item"
      key="3"
    >
      <newsComponent
        news={
          Object {
            "id": 3,
            "image": "https://cdn.deringhall.com/images/718/original/cover.jpg?auto=format&fit=max&q=60&w=670",
            "title": "Shining Through",
            "type": "Press",
          }
        }
      />
    </li>
    <li
      className="col dh-seller-profile-news-item"
      key="4"
    >
      <newsComponent
        news={
          Object {
            "id": 4,
            "image": "https://cdn.deringhall.com/images/681/original/cover.jpg?auto=format&fit=max&q=60&w=670",
            "title": "Remote Possibilities",
            "type": "Press",
          }
        }
      />
    </li>
  </ul>
</RelatedModuleTemplate>
`;
